gqltesting/testing.go

```diff
@@ -65,3 +67,28 @@ func formatJSON(t *testing.T, data []byte) []byte {
 	}
 	return formatted
 }
+
+func checkErrors(t *testing.T, expected, actual []*errors.QueryError) {
+	expectedCount, actualCount := len(expected), len(actual)
+
+	if expectedCount != actualCount {
+		t.Fatalf("unexpected number of errors: got %d, want %d", expectedCount, actualCount)
+	}
+
+	if expectedCount > 0 {
+		for i, want := range expected {
+			got := actual[i]
+
+			if !reflect.DeepEqual(got, want) {
```

Comment:
What do you think about checking that the output of their `Error()` methods are the same instead of `reflect.DeepEqual`?

Is there something in particular we're looking for with DeepEqual?

Reply:
Hmmm, I'm not quite sure. When it comes to error checking, I've always done `reflect.DeepEqual` rather than checking against strings. I guess for checking in `DeepEqual`, it'd be asserting things more granularly based on where they occurred for the errors rather than just the message of it. It was a simple error like `errors.New()` I could be inclined to compare `Error()` output. But looking at it, it could be a bit overkill to have to construct the `Location` struct for the query errors as well.

Reply:
Ah `Location` may make sense for validation errors. I think this is probably okay for now; we can always simplify later.

Reply:
Awesome. Is this ready to ship then?